src/instant/instantTransactions.ts

```diff
@@ -47,14 +49,29 @@ export function tradePayWithETH(
     calls.tradePayWithETHNoProxy({ ...state, ...gasData } as InstantOrderData);
 
   return tx$.pipe(
-    switchMap((txState: TxState) =>
-      of(progressChange({
+    switchMap((txState: TxState) => {
+
+      if (txState.status === TxStatus.Success) {
+        return of(progressChange({
+          ...initialProgress,
+          tradeTxStatus: txState.status,
+          tradeTxHash: getTxHash(txState),
+          summary: {
+            soldToken: txState.meta.args.sellToken,
+            boughtToken: txState.meta.args.buyToken,
+            ...extractTradeSummary(txState.receipt.logs),
```

Comment:
I don't like the duplicity of the code in both `doTradePayWithEth` and `tradePayWithERC20`.
I've tried to make the method to return the summary itself but I had to check inside of it if the `txState.status` was successful. Additionally I had to return null object which was complicating the code. 

Suggestions are accepted! 

Tried to put it into the pipeline somehow..

Reply:
It would not be that bad if result of ...extractTradeSummary(txState.receipt.logs) would be merged with progress directly. SoldToken, boughtToken are not necessary.